ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/ruleeditorhook.js

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright 2022 Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+(function() {
+
+    "use strict";
+    window.CQ = window.CQ || {};
+    window.CQ.FormsCoreComponents = window.CQ.FormsCoreComponents || {};
+    window.CQ.FormsCoreComponents.editorhooks =  window.CQ.FormsCoreComponents.editorhooks || {};
+    window.CQ.FormsCoreComponents.editorhooks.openRuleEditor = function (editable) {
+        let existingRuleEditorFrame = document.getElementById('af-rule-editor');
+        if (existingRuleEditorFrame) {
+            existingRuleEditorFrame.parentNode.removeChild(existingRuleEditorFrame);
+        }
+        let ruleEditorFrame = document.createElement('iframe');
+        ruleEditorFrame.setAttribute('id','af-rule-editor');
+        let ruleEditorUri = '/aem/af/expeditor.html' + getFormContainerPath(editable) + "?fieldPath="+ editable.path + "&fieldId="+ getFieldId(editable);
+        ruleEditorFrame.setAttribute('src', ruleEditorUri);
+        ruleEditorFrame.setAttribute('title', 'AF Rule Editor');
+        ruleEditorFrame.style.display = "block";
+        ruleEditorFrame.style.width = "100%";
+        ruleEditorFrame.style.height = "100%";
+        ruleEditorFrame.style.position = "fixed";
+        document.body.appendChild(ruleEditorFrame);
```

Comment:
If the frame exists, just changing its src attribute doesn't update the UI ? or we need to create it every time

Reply:
What happens if the data attribute doesn't exist on the node. We should throw an error or fallback to something. Not a blocker but we should define it and add a todo here

Reply:
> If the frame exists, just changing its src attribute doesn't update the UI ? or we need to create it every time

usually, iframe is removed as we do it explicitly in close handler.